extensions/wikia/TemplateClassification/TemplateClassification.hooks.php

```diff
@@ -15,7 +14,96 @@ public static function register() {
 		\Hooks::register( 'BeforePageDisplay', [ $hooks, 'onBeforePageDisplay' ] );
 		\Hooks::register( 'PageHeaderPageTypePrepared', [ $hooks, 'onPageHeaderPageTypePrepared' ] );
 		\Hooks::register( 'QueryPageUseResultsBeforeRecache', [ $hooks, 'onQueryPageUseResultsBeforeRecache' ] );
+		/* Edit page hooks */
+		\Hooks::register( 'EditPage::showEditForm:fields', [ $hooks, 'onEditPageShowEditFormFields' ] );
+		\Hooks::register( 'ArticleInsertComplete', [ $hooks, 'onArticleInsertComplete' ] );
 		\Hooks::register( 'EditPageLayoutExecute', [ $hooks, 'onEditPageLayoutExecute' ] );
+		\Hooks::register( 'EditPageMakeGlobalVariablesScript', [ $hooks, 'onEditPageMakeGlobalVariablesScript' ] );
+	}
+
+	/**
+	 * Save template type passed from article creation
+	 * template type is stored in templateClassificationType hidden field
+	 *
+	 * @param \WikiPage $wikiPage
+	 * @return bool
+	 */
+	public function onArticleInsertComplete( \WikiPage $wikiPage ) {
+		global $wgCityId;
+
+		$request = \RequestContext::getMain()->getRequest();
+		$type = $request->getVal( 'templateClassificationType' );
+
+		/**
+		 * The service was not available when the field's value was set
+		 * so we exit early to prevent polluting of the results.
+		 */
+		if ( $type === \TemplateClassificationService::NOT_AVAILABLE ) {
+			return true;
+		}
+
+		try {
+			( new \TemplateClassificationService() )->classifyTemplate(
+				$wgCityId,
+				$wikiPage->getId(),
+				$type,
+				\TemplateClassificationService::USER_PROVIDER,
+				$wikiPage->getUser()
+			);
+		} catch ( \Exception $e ) {
```

Comment:
Do we have to catch a generic `Exception`? Can we be more specific or is the API client throwing generic exceptions somewhere? Ideally we want to catch specific exceptions that we're expecting to occur if the service is unavailable, and other unexpected exceptions that occur in the called method should fail loudly so we find them and either fix them or also handle them. The same goes for the other catch block.s